app/js/lib/foxbox/network.js

```diff
@@ -49,43 +45,38 @@ export default class Network extends EventDispatcher {
    * @return {Promise}
    */
   init() {
-    window.addEventListener('online', this[p.pingAllBoxes]);
-    window.addEventListener('offline', this[p.pingAllBoxes]);
+    window.addEventListener('online', this[p.pingLinks]);
+    window.addEventListener('offline', this[p.pingLinks]);
 
-    let pingInterval;
     if ('connection' in navigator && 'onchange' in navigator.connection) {
-      navigator.connection.addEventListener('change', this[p.pingAllBoxes]);
+      navigator.connection.addEventListener('change', this[p.pingLinks]);
 
       // We also ping the box every few minutes to make sure it's still there.
-      pingInterval = this[p.settings].onlineCheckingLongInterval;
+      this[p.linkPingInterval] = this[p.settings].onlineCheckingLongInterval;
     } else {
       // If the Network Information API is not implemented, fallback to polling.
-      pingInterval = this[p.settings].onlineCheckingInterval;
+      this[p.linkPingInterval] = this[p.settings].onlineCheckingInterval;
     }
 
-    // @todo Settings should emit event when it changes "tunnelOrigin" or
-    // "localOrigin" setting and we should listen for this change and start or
-    // stop timers according to these settings.
-    this[p.localPingTimer] = new SequentialTimer(pingInterval);
-    this[p.localPingTimer].start(this[p.pingLocalBox]);
+    this[p.settings].on('local-origin', this[p.onLinkOriginChange]);
+    this[p.settings].on('remote-origin', this[p.onLinkOriginChange]);
 
-    this[p.remotePingTimer] = new SequentialTimer(pingInterval);
-    this[p.remotePingTimer].start(this[p.pingRemoteBox]);
+    this[p.onLinkOriginChange]();
+    this[p.onLinkOnlineChange]();
 
-    this[p.pingAllBoxes]();
+    this[p.pingLinks]();
 
     return Promise.resolve();
   }
 
   get origin() {
-    if (this[p.local]) {
-      return this.localOrigin;
-    } else if (this[p.remote]) {
-      return this.tunnelOrigin;
+    if (this[p.localLink]) {
+      return this[p.localLink].origin;
+    } else if (this[p.tunnelLink]) {
+      return this[p.tunnelLink].origin;
     }
 
-    console.error('The box is out of reach.');
-    return this.localOrigin;
+    throw new Error('Origin is not accessible');
```

Comment:
I don't have strong opinion here, but it seems if anything tries to access origin before we have it we should throw